[datletu] ip

datletu opened the pull request:
## KunkkaPro
> Step lively now, your Admiral is on board! - Kunkka ([source](https://dota2.fandom.com/wiki/Kunkka/Responses))

KunkkaPro keeps track of all the tasks you need to do. It's
+ text-based
+ easy to learn
+ ~~FAST~~ SUPER FAST to use 

All you need to do is
1. download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ)
2. double-click it
3. add your tasks
4. let it manages your tasks for you :shipit:

And it is **FREE**

Features
- [x] adding task
- [x] finding tasks
- [ ] ai assistant (coming soon)  

If you're a Java pro you can use the program to practice Java too. Here's the `main` method
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

datletu committed:
Level 0. Rename, Greet, Exit

datletu committed:
Level 1. Echo

datletu committed:
Level 2. Add, List and add a logo

datletu committed:
.

datletu committed:
Level 3. Mark as Done

datletu committed:
Level 4. ToDos, Events, Deadlines

datletu committed:
A-TextUiTesting

datletu committed:
Level-5

datletu committed:
Level-6

datletu committed:
Level-7

datletu committed:
add dates

datletu committed:
make exception  when from is after to

datletu committed:
.

datletu committed:
move data file to root

datletu committed:
Merge branch 'branch-Level-7'

datletu committed:
.

datletu committed:
Merge branch 'master' of https://github.com/datletu/ip into A-MoreOOP
.

datletu committed:
A-MoreOOP: Add more OOP features

datletu committed:
move to kunkka package

datletu committed:
.

datletu committed:
.

datletu committed:
Merge branch 'master' into A-Gradle
merge commit

datletu committed:
add gradle

datletu committed:
add checkstyle plugin

datletu committed:
add test cases for Tasklist and EventCommand

datletu committed:
make jar file

datletu committed:
add JavaDoc

datletu committed:
change all import to match coding standard

datletu committed:
.

datletu committed:
Merge branch 'master' of https://github.com/datletu/ip into Level-9-Find
.

datletu committed:
add find feature

ToxOptimism commented on src/main/java/kunkka/command/Command.java:
```diff
@@ -0,0 +1,17 @@
+package kunkka.command;
+import kunkka.tasklist.Tasklist;
```
Perhaps a line  break can be added here to separate packages and imports?

I noticed a couple areas with the same issue.

iamanoob44 replied:
Yes i agree with ToxOptimism.

Perhaps you should leave a line break to separate the package and the relevant imports?

I noticed couple of other files with the same issue.

kea-S replied:
Same here on my side! Best practice is to separate package and import statements with a line break. But putting the package statement at the very top of the file

ToxOptimism commented on src/main/java/kunkka/command/Command.java:
```diff
@@ -0,0 +1,17 @@
+import kunkka.tasklist.Tasklist;
+
+public abstract class Command {
+    protected String type;
```
Is there a reason why a type variable has been specified rather than checking the instance of an object?

Not having a type variable might help to reduce duplicate constructors.

ToxOptimism commented on src/main/java/kunkka/command/Command.java:
```diff
@@ -0,0 +1,17 @@
+package kunkka.command;
+import kunkka.tasklist.Tasklist;
+
+public abstract class Command {
```
Perhaps JavaDocs could be added to the classes to better understand the nature of the data type? This also ensures that it follows the coding standards of the project. 

I noticed the same issue in several other places too.

ToxOptimism commented on src/main/java/kunkka/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+
+    /**
+     * Constructor for DeadlineCommand
+     * @param description Description of the deadline task
```
Perhaps the description of the method and the params tagged could be separated? This ensures that it follows the JavaDoc standards. 

I noticed the same issue in several other places too.

ToxOptimism commented on src/main/java/kunkka/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+        try {
+            if (description.equals("")) {
+                throw new KunkkaException("Error: Task name cannot be empty");
+            }
+            else if (by.equals("")) {
```
It looks like there may be a small formatting issue here, shouldn't this be formatted as `} else if` to follow the project's coding style instead?

I noticed the same issue in several other places too.

ToxOptimism commented on src/main/java/kunkka/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+                System.out.println("  " + task);
+                System.out.println("Now you have " + tasks.getTasks().size() + " tasks in the list.");
+            }
+        }
+        catch (KunkkaException e) {
```
It looks like there may be a small formatting issue here, shouldn't this be formatted as `} catch` to follow the project's coding style instead?

I noticed the same issue in several other places too.

ToxOptimism commented on src/main/java/kunkka/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+import kunkka.components.Deadline;
+import kunkka.tasklist.Tasklist;
+
+public class DeadlineCommand extends Command {
```
Perhaps JavaDocs could be added to the classes to better understand the nature of the data type? This also ensures that it follows the coding standards of the project.

I noticed the same issue in several other places too.

ToxOptimism replied:
Perhaps naming the class `AddDeadlineCommand` could help to add some clarity of what the command does? 

It would still be a noun due to the word `Command`.

ToxOptimism commented on src/main/java/kunkka/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,47 @@
+package kunkka.command;
```
Perhaps a line break can be added here to separate packages and imports?

I noticed a couple areas with the same issue.

ToxOptimism commented on src/main/java/kunkka/command/Delete.java:
```diff
@@ -0,0 +1,37 @@
+ * Represents a delete command.
+ */
+public class Delete extends Command {
+    protected int index;
```
Perhaps a line break could be added here just to separate the class declaration and attributes?

ToxOptimism commented on src/main/java/kunkka/command/Find.java:
```diff
@@ -0,0 +1,35 @@
+ * Represents a command to find tasks with a keyword.
+ */
+public class Find extends Command {
+    private String keyword;
```
Perhaps a line break could be added here just to separate the class declaration and attributes?

ToxOptimism commented on src/main/java/kunkka/command/Find.java:
```diff
@@ -0,0 +1,35 @@
+/**
+ * Represents a command to find tasks with a keyword.
+ */
+public class Find extends Command {
```
Perhaps change the name of the class to FindCommand such that the class name isnt a verb?

Appending Command to the end of the class name might also help indicate its relation to the other commands.

I have noticed several other classes with the same issue.

kea-S replied:
I think I differ in ToxOptimism's opinion here.

IIRC class names are better off named as nouns, so maybe CommandFinder is a better class name here.

ToxOptimism replied:
Oh thats my bad, I meant to say that `FindCommand`, a noun might be a more appropriate class name than `Find` a verb. 

`CommandFinder` would however imply that the class finds commands rather than a command that handles finding, which I believe is not the intended meaning here.

ToxOptimism commented on src/main/java/kunkka/command/List.java:
```diff
@@ -0,0 +1,22 @@
+/**
+ * Represents the command to list all tasks in the task list.
+ */
+public class List extends Command {
```
Perhaps change the name of the class to ListCommand to group the name of the class with the rest of the commands?

ToxOptimism commented on src/main/java/kunkka/command/EventCommand.java:
```diff
@@ -0,0 +1,60 @@
+/**
+ * Represents a command to add an event task to the task list.
+ */
+public class EventCommand extends Command {
```
Perhaps naming the class AddEventCommand could help to add some clarity of what the command does?

It would still be a noun due to the word Command.

I have noticed several other areas with similar issues.

ToxOptimism commented on src/main/java/kunkka/components/Deadline.java:
```diff
@@ -0,0 +1,45 @@
+        this.by = LocalDate.parse(by);
+    }
+
+    /**
+     * Constructor for Deadline.
+     *
+     * @param name Name of the deadline task.
+     * @param by Deadline of the task.
+     */
```
I believe this is the incorrect JavaDoc for this method.

kea-S replied:
I concur

ToxOptimism commented on src/main/java/kunkka/components/Event.java:
```diff
@@ -0,0 +1,73 @@
+
+    /**
+     * Returns the start date of the event.
+     * @return Start date of the event.
```
Perhaps a new line could be added between the JavaDoc description and the @return tag to follow JavaDocs convention?

I have noticed several areas which also face a similar issue.

ToxOptimism commented on src/main/java/kunkka/components/Task.java:
```diff
@@ -0,0 +1,78 @@
+public class Task {
+    protected String name;
+    protected boolean isDone;
+    protected String type;
```
Is there a reason why a type variable has been specified rather than checking the instance of an object?

Not having a type variable might help to unnecessary constructor declaration in subclasses.

ToxOptimism commented on src/main/java/kunkka/mainChatBot/Kunkka.java:
```diff
@@ -0,0 +1,47 @@
+package kunkka.mainChatBot;
```
Perhaps name this package as `chatbot`?

Packages should be written in all lower case not camel case.

Alternatively perhaps the main application code can be placed under the `kunkka` package directly?

ToxOptimism commented on src/main/java/kunkka/components/Task.java:
```diff
@@ -0,0 +1,78 @@
+     * @param type Type of the task.
+     * @param isDone Status of the task.
+     */
+    public Task(String name, String type, boolean isDone) {
```
Is there a reason as to why `isDone` is a parameter of the constructor? It seems that the default value of `isDone` is false regardless of what add task command is used.

ToxOptimism commented on src/main/java/kunkka/components/Event.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    @Override
+    public String toString() {
+        return "[E]" + "[" + getStatusIcon() + "] " + name + " (from: " + getFromFormatted() + " to: " + getToFormatted() + ")";
```
Is there any reason why it does not use `[E] + super.toString() + " (from: " ...` such that it builds off the original toString?

I have noticed several other areas with similar issues.

iamanoob44 replied:
Is there any reason as to why you did not invoke a `super.toString()` method? Task is the parent class of all other task types like `Deadline` , `Event` and `Todo`.

I have noticed a similar issue for these other classes too, that are subtypes of `Task`.

ToxOptimism commented on src/main/java/kunkka/components/Todo.java:
```diff
@@ -0,0 +1,21 @@
+
+    @Override
+    public String toString() {
+        return "[T]" + "[" + getStatusIcon() + "] " + name;
```
Is there any reason why it does not use `[T] + super.toString() + " (from: " ...` such that it builds off the original toString, or simply `super.toString()`

I have noticed several other areas with similar issues.

ToxOptimism commented on src/main/java/kunkka/mainChatBot/Kunkka.java:
```diff
@@ -0,0 +1,47 @@
+        sc.close();
+        
+        //Save tasks to file
+        storage.save(tasks);
```
Is there any reason why the tasks are saved here, I believe that according to the requirements, the tasks should be saved whenever you mark, unmark, add, or delete a task?

ToxOptimism commented on src/main/java/kunkka/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+ */
+public class Parser {
+    /**
+     * Parses a task from a string
```
I believe that this JavaDoc is meant to indicate parse from file?
Perhaps the JavaDoc description could be made more clear instead?

ToxOptimism commented on src/main/java/kunkka/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+        //Handle list command
+        if (command.equals("list")) {
+            return new List();
+        } 
```
It looks like there may be a small formatting issue here, shouldn't this be formatted as `} else if` to follow the project's coding style instead?

I noticed the same issue in several other places too.

ToxOptimism commented on src/main/java/kunkka/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+
+        //Handle invalid command
+        else {
+            try {
```
Is there any reason why the try catch is done here?

I believe the Kunkka.java already catches such an exception. Though understandably it appears that its more so for executing the command rather than creating the command. 

However since the try catch returns null, the Kunkka code might catch an exception due to null.execute() instead.

ToxOptimism commented on src/main/java/kunkka/mainChatBot/Kunkka.java:
```diff
@@ -0,0 +1,47 @@
+            try {
+                Command c = Parser.parseCommand(command.trim());
+                c.execute(tasks);
+            } catch (Exception e) {
```
Perhaps catch `KunkkaException` rather than all exceptions to avoid cases where an unexpected exception was caught and displayed to the user?

ToxOptimism commented on src/main/java/kunkka/tasklist/Tasklist.java:
```diff
@@ -0,0 +1,134 @@
+     * @param index Index of the task to be deleted.
+     */
+    public void deleteTask(int index) {
+        try {
```
Is there a reason why the exception handling is done here rather than at `Delete`'s execute() like how the exceptions for adding tasks was done?

ToxOptimism commented on src/main/java/kunkka/tasklist/Tasklist.java:
```diff
@@ -0,0 +1,134 @@
+     * @param keyword Keyword to search for.
+     */
+    public void markTaskAsDone(int index) {
+        try {
```
Is there a reason why the exception handling is done here rather than at `Mark`'s execute() like how the exceptions for adding tasks was done?

I have noticed several areas with the same issues.

ToxOptimism commented on src/test/java/kunkka/command/EventCommandTest.java:
```diff
@@ -0,0 +1,112 @@
+public class EventCommandTest {
+
+    @Test
+    public void testValidEvent() {
```
Perhaps write test cases declaration as function_testInput_expectedOutput per the standards for JUnit?

I have noticed several areas with the same issue.

ToxOptimism commented on src/test/java/kunkka/components/TodoTest.java:
```diff
@@ -0,0 +1,15 @@
+public class TodoTest {
+    
+    @Test
+    public void testToString() {
```
Perhaps write test cases declaration as function_testInput_expectedOutput per the standards for JUnit?

I have noticed several areas with the same issue.

ToxOptimism reviewed (commented):
Overall it is quite well done, there are some nit regarding missing JavaDocs and issues with coding style, standards, package and method naming however. Additionally, there are certain parts of the code in which I am unsure why it was coded in such a way.

iamanoob44 commented on src/main/java/kunkka/command/Command.java:
```diff
@@ -0,0 +1,17 @@
+        return type;
+    }
+
+    public abstract void execute(Tasklist tasks);
```
Perhaps you can include 1 or 2 JavaDoc comments for the methods/constructor to better understand how each of them works? This would also ensures it follows more closely to the coding standards of the course.

iamanoob44 commented on src/main/java/kunkka/command/Delete.java:
```diff
@@ -0,0 +1,37 @@
+     * 
+     * @param tasklist Tasklist to execute the command on.
+     */
+    public void execute(Tasklist tasklist) {
```
For line 34, perhaps you should rename the parameter `tasklist` as `taskList` to follow the coding standards.

Overall, still looks good to me! JavaDoc comments included are informative and I do not see any blatant violation of the coding standard here, and I like how this bit of code is clean.

iamanoob44 commented on src/main/java/kunkka/command/EventCommand.java:
```diff
@@ -0,0 +1,60 @@
+        this.to = to;
+    }
+
+    /**
+     * Executes the event command to add an event task to the task list.
+     * 
+     * @param tasks Tasklist object that stores the list of tasks.
+     */
```
Perhaps you can include `@throw KunkkaException for invalid inputs` for a more complete JavaDoc comment?

I noticed the same issue for other places too.

iamanoob44 commented on src/main/java/kunkka/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+import kunkka.components.Event;
+import kunkka.components.KunkkaException;
+
+import kunkka.command.Command;
+import kunkka.command.List;
+import kunkka.command.Mark;
+import kunkka.command.Unmark;
+import kunkka.command.TodoCommand;
+import kunkka.command.DeadlineCommand;
+import kunkka.command.EventCommand;
+import kunkka.command.Delete;
+import kunkka.command.Find;
```
Perhaps you can reorganise your imports so that it is organised alphabetially? This way, it conforms to the coding standard for the course.

iamanoob44 commented on src/main/java/kunkka/storage/Storage.java:
```diff
@@ -0,0 +1,84 @@
+            FileWriter fileWriter = new FileWriter("./data/kunkka.txt");
+            List<Task> tasks = taskslist.getTasks();
+            for (Task task : tasks) {
+                fileWriter.write(task.getType() + " | " + (task.getIsDone() ? "1" : "0") + " | " + task.getName());
```
Perhaps you can move the some chunks of the code to the next bottom line, line 72?

This way, we will be more certain that line 71 would not exceed 120 characters, which would violate the coding standard

iamanoob44 commented on src/main/java/kunkka/parser/Parser.java:
```diff
@@ -0,0 +1,111 @@
+        //Handle todo command
+        else if (command.matches("todo .*")) {
+            return new TodoCommand(command.split(" ", 2)[1]);
+        }
```
Perhaps you can move the right curly bracket `}` such that it is on the same line as the `else if` statement?
====> `} else if ... { ...` 

I noticed the same issue for several areas too.

This way, it follows more closely to the 'Egyptian Style' braces convention.

iamanoob44 commented on src/main/java/kunkka/components/Task.java:
```diff
@@ -0,0 +1,78 @@
+package kunkka.components;
```
Overall, looks good here! There is no serious blatant violations of the coding standard.

JavaDoc comments for most of the methods are also included too

iamanoob44 reviewed (commented):
Overall, there are parts of the code that conform to the coding standard, so great work on that! However, there are also some parts of the code that have to be tweaked to align with the coding standard.

kea-S commented on src/main/java/kunkka/components/KunkkaException.java:
```diff
@@ -0,0 +1,10 @@
+package kunkka.components;
+
+/**
+ * KunkkaException
```
Perhaps more details could be given in the JavaDoc comments here

kea-S commented on src/main/java/kunkka/storage/Storage.java:
```diff
@@ -0,0 +1,84 @@
+
+    /**
+     * Load tasks from file.
+     * @return Tasklist containing tasks loaded from file.
```
Mayhaps you can consider adding linking words Tasklist here to the actual Tasklist class. Or maybe use the see also annotation

kea-S reviewed (commented):
Overall good job just can be clearer on some things, keep going!

datletu commented:
Hi so sorry for the late reply. I just saw your comments. Thank you for
your feedback. I will implement the changes right away.

On Wed, Feb 5, 2025 at 4:40 PM ToxicOptimism ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/main/java/kunkka/command/Find.java
> <https://github.com/nus-cs2103-AY2425S2/ip/pull/193#discussion_r1942450403>
> :
>
> > @@ -0,0 +1,35 @@
> +package kunkka.command;
> +import kunkka.tasklist.Tasklist;
> +import kunkka.components.Task;
> +
> +/**
> + * Represents a command to find tasks with a keyword.
> + */
> +public class Find extends Command {
>
> Oh thats my bad, I meant to say that FindCommand, a noun would be a
> better class name compared to Find a verb.
>
> CommandFinder would however imply that the class finds commands rather
> than a command that handles finding, which I believe is not the intended
> meaning here.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nus-cs2103-AY2425S2/ip/pull/193#discussion_r1942450403>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMO3GK5J24FQJBMBLYHN6WD2OHFAXAVCNFSM6AAAAABWAEGO6KVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKOJUHE4DENZYGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

datletu committed:
copy the tutorial

datletu committed:
implemented gui

datletu committed:
added bye command

datletu committed:
Addition of assertions to the codebase.

There are several points where assertions can be added to the codebase to ensure that the code is functioning as expected. This commit adds assertions to the codebase to ensure that the code is functioning as expected.

As a result, the codebase is more robust and less prone to errors. This commit also adds a new test case to test the assertions added to the codebase.

datletu committed:
Merge pull request #2 from datletu/Branch-A-Assertions

Addition of assertions to the codebase.

datletu committed:
refraction: Refactor code quality

split up the parser functions into smaller functions for better readability

ensure that all functions do not exceed 30 LoCs

datletu committed:
Merge pull request #3 from datletu/branch-A-CodeQuality

refraction: Refactor code quality

datletu committed:
Implement Priority feature

Allow users to set priority for tasks and filter tasks by priority

Added commands:

1. setpriority <index> <priority>

2. findpriority <priority>

priority is an arbitrary integer value

datletu committed:
.

datletu committed:
Merge branch 'master' of https://github.com/datletu/ip

datletu committed:
Fix an indexing bug for SetPriority command

There is a bug where the index shown in the response message for
Setpriority was the array index and not 1-indexing

As a step towards unification, let's make all indexing in response messages
1-indexed

datletu committed:
Add a new message when Find command does not find
any task that fit the description.

Give the error message "Invalid command" when the
priority is not specified when adding a task

datletu committed:
Add a title to the title bar of the application window

datletu committed:
Add UI screenshot

datletu committed:
Update README.md

Add user guide for the app

datletu committed:
Fix bug in storage and parser where the formats of the stored
files and the parser are inconsistent.

Fix bugs in delete command where the deletion takes place before the
response message, causing error.

Make a new response message for set prirority command
to show the task name instead of index for more readability for users.

datletu committed:
Merge branch 'master' of https://github.com/datletu/ip

datletu committed:
change java version to 17

datletu committed:
add greeting line

make priority no longer compulsory
